vxquery-core/src/main/java/org/apache/vxquery/runtime/functions/json/ValueScalarEvaluator.java

```
@@ -0,0 +1,83 @@
+import org.apache.vxquery.exceptions.SystemException;
+import org.apache.vxquery.runtime.functions.base.AbstractTaggedValueArgumentScalarEvaluator;
+
+public class ValueScalarEvaluator extends AbstractTaggedValueArgumentScalarEvaluator {
+    protected final IHyracksTaskContext ctx;
+    private final ArrayBackedValueStorage mvs;
+    private final ArrayPointable ap;
+    private final LongPointable lp;
+    private final SequenceBuilder ab;
+    protected DataOutput out;
+
+    public ValueScalarEvaluator(IHyracksTaskContext ctx, IScalarEvaluator[] args) {
+        super(args);
+        this.ctx = ctx;
+        ap = (ArrayPointable) ArrayPointable.FACTORY.createPointable();
+        lp = (LongPointable) LongPointable.FACTORY.createPointable();
+        mvs = new ArrayBackedValueStorage();
+        ab = new SequenceBuilder();
+    }
+
+    @Override
+    protected void evaluate(TaggedValuePointable[] args, IPointable result) throws SystemException {
+        mvs.reset();
+        try {
+            ab.reset(mvs);
+            TaggedValuePointable tvp1 = args[0];
+            if (tvp1.getTag() == ValueTag.ARRAY_TAG) {
+                tvp1.getValue(ap);
+                TaggedValuePointable tvp2 = args[1];
+                TaggedValuePointable tempTvp = ppool.takeOne(TaggedValuePointable.class);
+                tvp2.getValue(lp);
+                if ((int) lp.getLong() > ap.getEntryCount())
+                    throw new RuntimeException("The argument is out of bounds");
+                ap.getEntry((int) lp.getLong() - 1, tempTvp);
+                try {
+                    if (ap.getEntryCount() != 0)
+                        ab.addItem(tempTvp);
+                    else
+                        ab.addItem(tvp1);
+                } finally {
+                    ppool.giveBack(tempTvp);
+                }
+                ab.finish();
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
```

Please don't swallow exceptions. It seems that we won't get the correct result here and so we should abort. One option would be to wrap the exception into a SystemException with error code SYSE0001.